File: Java-SpringBoot/BE-02-DB/java-17-springboot-3.5/src/main/java/com/flyrank/taskservice/service/TaskService.java
package com.flyrank.taskservice.service;

import com.flyrank.taskservice.dto.*;

import java.util.List;

/**
 * Service interface for Task business operations.
 */
public interface TaskService {
    List<TaskResponse> getAllTasks(String search, Boolean done);
    TaskResponse getTaskById(Long id);
    TaskResponse createTask(TaskCreateRequest request);
    TaskResponse replaceTask(Long id, TaskReplaceRequest request);
    TaskResponse patchTask(Long id, TaskUpdateRequest request);
    void deleteTask(Long id);
    StatsResponse getStats();
}

File: Java-SpringBoot/BE-02-DB/kotlin-2.3-springboot-3.5/src/main/kotlin/com/flyrank/taskservice/service/TaskService.kt
package com.flyrank.taskservice.service

import com.flyrank.taskservice.dto.*

interface TaskService {
    fun getAllTasks(search: String?, done: Boolean?): List<TaskResponse>
    fun getTaskById(id: Long): TaskResponse
    fun createTask(request: TaskCreateRequest): TaskResponse
    fun replaceTask(id: Long, request: TaskReplaceRequest): TaskResponse
    fun patchTask(id: Long, request: TaskUpdateRequest): TaskResponse
    fun deleteTask(id: Long)
    fun getStats(): StatsResponse
}
